### PR TITLE
Docs: fix Pull Request Process in CONTRIBUTING.md

### LINKED ISSUE #2651: Docs: Pull Request Process described in CONTRIBUTING.md doesn't work as expected
The Pull Request Process as described in CONTRIBUTING.md gives us following steps:
```
$ mkdir -p $GOPATH/src/github.com/thanos-io
$ cd $GOPATH/src/github.com/thanos-io
$ git clone https://github.com/<your_github_id>/thanos.git
$ cd thanos
$ git remote add upstream https://github.com/thanos-io/thanos.git
$ git remote update
$ git merge upstream/master
$ make build
$ ./thanos -h
```

There are some minor problems here.

```
$ mkdir -p $GOPATH/src/github.com/thanos-io
```
In most of the cases, the environment variable GOPATH isn't set, so this command tries to create a folder at `/src/github.com/thanos-io` which is not what we want.
A possible solution is to run `GOPATH=$(go env GOPATH)` first.

Also, as we use Go modules we actually don't need to clone the repo inside the GOPATH, it works fine irrespective of the folder's location.

```
$ make build
$ ./thanos -h
```
Here, `./thanos -h` fails because after #2611 the `make build` outputs the `thanos` binary in `$GOBIN` instead of the current folder. Maybe we can run this using `$GOPATH/bin/thanos -h` but I guess the proper way would be to have the `$GOPATH/bin` directory in the `PATH` variable.

### PR DESCRIPTION
**Pull request by prmsrswt**
Signed-off-by: Prem Kumar <prmsrswt@gmail.com>

* [ ] I added CHANGELOG entry for this change.
* [x] Change is not relevant to the end user.

## Changes
Fixes #2651

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -107,7 +108,8 @@ $ git remote add upstream https://github.com/thanos-io/thanos.git
 $ git remote update
 $ git merge upstream/master
 $ make build
-$ ./thanos -h
+$ export PATH=$PATH:$GOPATH/bin
+$ thanos -h
```
Comment (soniasingla) [inline CONTRIBUTING.md:112]: @prmsrswt `thanos -h` could be replaced with `thanos -h flags/command/args` 
Also, additionally we can link it to `scripts/quickstart.sh` to get the basic idea of which components to run :) 

WDYT @daixiang0 @bwplotka ?

Comment (bwplotka) [reply CONTRIBUTING.md:112]: > thanos -h could be replaced with thanos -h flags/command/args

What is ` thanos -h flags/command/args`? :thinking: 

> Also, additionally we can link it to scripts/quickstart.sh to get the basic idea of which components to run :)

Yes!

Comment (soniasingla) [reply CONTRIBUTING.md:112]: > > thanos -h could be replaced with thanos -h flags/command/args
> 
> What is ` thanos -h flags/command/args`? 🤔
 
We need to mention the arguments/flags/commands whichever we are using after `thanos -h` 
for ex : `sidecar` , `store` are inculded under commands :)

Comment (bwplotka) [reply CONTRIBUTING.md:112]: hmmm you can but what's wrong with printing all of them? (: `thanos -h` just works and essentially it's the way to test if you build Thanos properly for a start.

Comment (daixiang0) [reply CONTRIBUTING.md:112]: I think the sub command test would be done with `make docs`, as @bwplotka said, it is enough for a start.

**Pull request merged**